collector/mssql.go

```diff
@@ -1324,79 +1390,83 @@ type win32PerfRawDataSQLServerAvailabilityReplica struct {
 	SendstoTransportPersec         uint64
 }
 
-func (c *MSSQLCollector) collectAvailabilityReplica(ch chan<- prometheus.Metric, sqlInstance string) (*prometheus.Desc, error) {
+func mssqlCollectAvailabilityReplica(c *MSSQLCollector, ch chan<- prometheus.Metric) (*prometheus.Desc, error) {
```

Comment:
Is there a reason not to keep these new collect functions as bound to the MSSQLCollector struct?

Reply:
I couldn't figure out how to do that AND have the ability to pass the function reference as a parameter to the `mssqlExecute()` function.

If you can find an example of someone going that, I'd be happy to change this.

Reply:
Hm. Do you mean something like this? https://play.golang.org/p/HcgvRIBLCxe
Maybe I'm missing your point?

Reply:
thanks for that nudge.  I think I got it.